Submitted By: DJ Lucas <dj_At_linuxfromscratch_DOT_org>
Date: 2004-12-12
Origin: OOO-Build-1.3.7
Initial Package Version: 1.1.3
Description: Fixes build when used with gcc-3.3.3.
Upstream Status: Not needed

$LastChangedBy$
$Date$

--- OOo_1.1.3_src-orig/icu/makefile.mk	2003-07-16 12:26:10.000000000 -0500
+++ OOo_1.1.3_src/icu/makefile.mk	2004-12-19 00:10:13.591321672 -0600
@@ -93,7 +93,7 @@
 
 CONFIGURE_DIR=source
 
-CONFIGURE_ACTION=sh -c 'CFLAGS=-O CXXFLAGS=-O ./configure --enable-layout --enable-static --enable-shared=yes --enable-64bit-libs=no'
+CONFIGURE_ACTION=sh -c 'CFLAGS=-O CXXFLAGS=-O ./configure --enable-layout --enable-static --enable-shared=yes --enable-64bit-libs=no --enable-extras=no'
 
 #CONFIGURE_FLAGS=--enable-layout --enable-static --enable-shared=yes --enable-64bit-libs=no
 CONFIGURE_FLAGS=
diff -Naur OOo_1.1.3_src-orig/stlport/STLport-4.5.3.patch OOo_1.1.3_src/stlport/STLport-4.5.3.patch
--- OOo_1.1.3_src-orig/stlport/STLport-4.5.3.patch	2003-03-27 05:54:46.000000000 -0600
+++ OOo_1.1.3_src/stlport/STLport-4.5.3.patch	2004-12-15 22:26:18.068938808 -0600
@@ -1,29 +1,28 @@
-*** misc/STLport-4.5.3/src/common_rules.mak	Mon Oct  8 19:30:30 2001
---- misc/build/STLport-4.5.3/src/common_rules.mak	Fri May  3 18:44:50 2002
+*** misc/STLport-4.5.3/src/fstream.cpp	Tue Sep  4 19:10:12 2001
+--- misc/build/STLport-4.5.3/src/fstream.cpp	Fri Jan 11 17:22:40 2002
 ***************
-*** 84,90 ****
-  	$(RM) $(OUTDIR)/$(STLDEBUG_NAME).$(DYNEXT)
-  	$(RM) $(OUTDIR)/$(DEBUG_NAME).$(DYNEXT)
-  	ln -s $(RELEASE_DYNLIB) $(OUTDIR)/$(RELEASE_NAME).$(DYNEXT)
-! 	-ln -s $(DEBUG_DYNLIB) $(OUTDIR)/$(DEBUG_NAME).$(DYNEXT)
-  	ln -s $(STLDEBUG_DYNLIB) $(OUTDIR)/$(STLDEBUG_NAME).$(DYNEXT)
-  
-  install_unix :
---- 84,91 ----
-  	$(RM) $(OUTDIR)/$(STLDEBUG_NAME).$(DYNEXT)
-  	$(RM) $(OUTDIR)/$(DEBUG_NAME).$(DYNEXT)
-  	ln -s $(RELEASE_DYNLIB) $(OUTDIR)/$(RELEASE_NAME).$(DYNEXT)
-! # enable when needed
-! #	-ln -s $(DEBUG_DYNLIB) $(OUTDIR)/$(DEBUG_NAME).$(DYNEXT)
-  	ln -s $(STLDEBUG_DYNLIB) $(OUTDIR)/$(STLDEBUG_NAME).$(DYNEXT)
-  
-  install_unix :
-*** misc/STLport-4.5.3/src/gcc-3.0.mak	Fri May  3 18:43:29 2002
---- misc/build/STLport-4.5.3/src/gcc-3.0.mak	Fri May  3 17:59:41 2002
+*** 293,299 ****
+  #ifdef __MINGW32__
+   __MINGW_IMPORT ioinfo * __pioinfo[];
+  #else
+!   extern _CRTIMP ioinfo * __pioinfo[];
+  #endif
+  
+  } // extern "C"
+--- 293,299 ----
+  #ifdef __MINGW32__
+   __MINGW_IMPORT ioinfo * __pioinfo[];
+  #else
+!   extern __declspec( dllimport ) ioinfo * __pioinfo[];
+  #endif
+  
+  } // extern "C"
+*** misc/STLport-4.5.3/src/gcc-3.0.mak	Fri Jan 11 12:48:44 2002
+--- misc/build/STLport-4.5.3/src/gcc-3.0.mak	Fri Jan 11 12:47:21 2002
 ***************
 *** 1 ****
 ! dummy
---- 1,56 ----
+--- 1,59 ----
 ! #
 ! # Note : this makefile is for gcc-3 !
 ! #
@@ -31,6 +30,8 @@
 ! #
 ! # compiler
 ! #
+! CC = $(CCCOMP) 
+! CXX = $(CXXCOMP) -D_REENTRANT -fexceptions
 ! 
 ! #
 ! # Basename for libraries
@@ -42,7 +43,7 @@
 ! #
 ! #
 ! LINK=ar cr
-! DYN_LINK=g++ --fexceptions -shared -o
+! DYN_LINK=$(CXX) --fexceptions -shared -o
 ! 
 ! OBJEXT=o
 ! DYNEXT=so
@@ -59,12 +60,12 @@
 ! 
 ! WARNING_FLAGS= -Wall -W -Wno-sign-compare -Wno-unused -Wno-uninitialized -ftemplate-depth-32
 ! 
-! CXXFLAGS_COMMON = -I${STLPORT_DIR} ${WARNING_FLAGS} -D_REENTRANT -fexceptions
+! CXXFLAGS_COMMON = -I${STLPORT_DIR} ${WARNING_FLAGS}
 ! 
-! CXXFLAGS_RELEASE_static = $(CXXFLAGS_COMMON) -O2
+! CXXFLAGS_RELEASE_static = $(CXXFLAGS_COMMON) -O2 -fPIC
 ! CXXFLAGS_RELEASE_dynamic = $(CXXFLAGS_COMMON) -O2 -fPIC
 ! 
-! CXXFLAGS_DEBUG_static = $(CXXFLAGS_COMMON) -g
+! CXXFLAGS_DEBUG_static = $(CXXFLAGS_COMMON) -g -fPIC
 ! CXXFLAGS_DEBUG_dynamic = $(CXXFLAGS_COMMON) -g -fPIC
 ! 
 ! CXXFLAGS_STLDEBUG_static = $(CXXFLAGS_DEBUG_static) -D_STLP_DEBUG
@@ -80,40 +81,92 @@
 ! #%.s: %.cpp
 ! #	$(CXX) $(CXXFLAGS) -O4 -S -pto $<  -o $@
 ! 
-*** misc/STLport-4.5.3/src/sunpro-common.mak	Thu Nov 29 20:26:22 2001
---- misc/build/STLport-4.5.3/src/sunpro-common.mak	Fri May  3 17:59:41 2002
+! 
+*** misc/STLport-4.5.3/src/vc6.mak	Mon Jul 30 00:18:15 2001
+--- misc/build/STLport-4.5.3/src/vc6.mak	Fri Jan 11 17:41:08 2002
 ***************
-*** 45,50 ****
---- 45,53 ----
-  STLDEBUG_static_rep = -ptr${STLDEBUG_OBJDIR_static}
-  STLDEBUG_dynamic_rep = -ptr${STLDEBUG_OBJDIR_dynamic}
+*** 10,16 ****
+  COMP=VC6
   
-+ RELEASE_DYNLIB_SONAME=libstlport_sunpro.so
-+ STLDEBUG_DYNLIB_SONAME=libstlport_sunpro_stldebug.so
-+ 
-  CXXFLAGS_RELEASE_static = $(CXXFLAGS_COMMON) ${RELEASE_FLAGS} ${RELEASE_static_rep}
-  CXXFLAGS_RELEASE_dynamic = $(CXXFLAGS_COMMON) ${RELEASE_FLAGS} $(SHCXXFLAGS) ${RELEASE_dynamic_rep}
+  # EXTRA_COMMON_FLAGS=/D "_MBCS"
+! EXTRA_COMMON_FLAGS=/Zm800 /FI "vc_warning_disable.h" /D "_MBCS"
+  EXTRA_DEBUG_FLAGS=
+  EXTRA_NDEBUG_FLAGS=
+  
+--- 10,16 ----
+  COMP=VC6
+  
+  # EXTRA_COMMON_FLAGS=/D "_MBCS"
+! EXTRA_COMMON_FLAGS=/Zm800 /FI "vc_warning_disable.h" /D "_MBCS" /D "_NTSDK" /D "_CRTIMP="
+  EXTRA_DEBUG_FLAGS=
+  EXTRA_NDEBUG_FLAGS=
+  
+*** misc/STLport-4.5.3/src/vc_common.mak	Tue Sep  4 19:10:12 2001
+--- misc/build/STLport-4.5.3/src/vc_common.mak	Fri Jan 11 12:48:14 2002
+***************
+*** 62,68 ****
+  LDFLAGS_RELEASE=/opt:ref
+  
+  LDFLAGS_COMMON_static=$(LDFLAGS_COMMON)
+! LDFLAGS_COMMON_dynamic=$(LDFLAGS_COMMON) /dll /incremental:no
+  
+  
+  CXXFLAGS_DEBUG_static=$(FLAGS_COMMON_static) /MTd $(FLAGS_DEBUG) /Fo"$(DEBUG_OBJDIR_static)\\" /Fd"$(DEBUG_OBJDIR_static)\\" /YXstlport_prefix.h /Fp$(DEBUG_OBJDIR_static)\stlport.pch
+--- 62,68 ----
+  LDFLAGS_RELEASE=/opt:ref
+  
+  LDFLAGS_COMMON_static=$(LDFLAGS_COMMON)
+! LDFLAGS_COMMON_dynamic=$(LDFLAGS_COMMON) /nodefaultlib oldnames.lib kernel32.lib /dll /incremental:no
+  
+  
+  CXXFLAGS_DEBUG_static=$(FLAGS_COMMON_static) /MTd $(FLAGS_DEBUG) /Fo"$(DEBUG_OBJDIR_static)\\" /Fd"$(DEBUG_OBJDIR_static)\\" /YXstlport_prefix.h /Fp$(DEBUG_OBJDIR_static)\stlport.pch
+***************
+*** 79,87 ****
+  
+  
+  LDFLAGS_DEBUG_static=$(LDFLAGS_COMMON_static)
+! LDFLAGS_DEBUG_dynamic=$(LDFLAGS_COMMON_dynamic) $(LDFLAGS_DEBUG) /implib:"$(OUTDIR)\$(DEBUG_NAME).$(STEXT)"
+  LDFLAGS_STLDEBUG_static=$(LDFLAGS_COMMON_static)
+! LDFLAGS_STLDEBUG_dynamic=$(LDFLAGS_COMMON_dynamic) $(LDFLAGS_DEBUG) /implib:"$(OUTDIR)\$(STLDEBUG_NAME).$(STEXT)" 
   
-*** misc/STLport-4.5.3/stlport/config/stl_sunpro.h	Sat Feb  9 03:01:00 2002
---- misc/build/STLport-4.5.3/stlport/config/stl_sunpro.h	Fri May  3 17:59:41 2002
+  # LDFLAGS_DEBUG_static=$(LDFLAGS_COMMON_static)  /DEBUGTYPE:CV
+  # LDFLAGS_DEBUG_dynamic=$(LDFLAGS_COMMON_dynamic) /DEBUG /DEBUGTYPE:CV /implib:"$(OUTDIR)\$(DEBUG_NAME).$(STEXT)" 
+--- 79,87 ----
+  
+  
+  LDFLAGS_DEBUG_static=$(LDFLAGS_COMMON_static)
+! LDFLAGS_DEBUG_dynamic=$(LDFLAGS_COMMON_dynamic) msvcrtd.lib $(LDFLAGS_DEBUG) /implib:"$(OUTDIR)\$(DEBUG_NAME).$(STEXT)"
+  LDFLAGS_STLDEBUG_static=$(LDFLAGS_COMMON_static)
+! LDFLAGS_STLDEBUG_dynamic=$(LDFLAGS_COMMON_dynamic) msvcrtd.lib $(LDFLAGS_DEBUG) /implib:"$(OUTDIR)\$(STLDEBUG_NAME).$(STEXT)" 
+  
+  # LDFLAGS_DEBUG_static=$(LDFLAGS_COMMON_static)  /DEBUGTYPE:CV
+  # LDFLAGS_DEBUG_dynamic=$(LDFLAGS_COMMON_dynamic) /DEBUG /DEBUGTYPE:CV /implib:"$(OUTDIR)\$(DEBUG_NAME).$(STEXT)" 
 ***************
-*** 40,45 ****
---- 40,46 ----
-  #  define _STLP_NO_CLASS_PARTIAL_SPECIALIZATION 1
-  #  define _STLP_NO_MEMBER_TEMPLATE_CLASSES 1
-  #  define _STLP_USE_OLD_HP_ITERATOR_QUERIES
-+ #  define _STLP_NO_MEMBER_TEMPLATES 1
-  # endif 
+*** 95,101 ****
+  CXXFLAGS_RELEASE_dynamic=$(FLAGS_COMMON_dynamic) /MD $(FLAGS_NDEBUG) /Fo"$(RELEASE_OBJDIR_dynamic)\\" /Fd"$(RELEASE_OBJDIR_dynamic)\\" /YXstlport_prefix.h /Fp$(RELEASE_OBJDIR_dynamic)\stlport.pch
+  
+  LDFLAGS_RELEASE_static=$(LDFLAGS_COMMON_static)
+! LDFLAGS_RELEASE_dynamic=$(LDFLAGS_COMMON_dynamic) $(LDFLAGS_RELEASE) /implib:"$(OUTDIR)\$(RELEASE_NAME).$(STEXT)" 
+  # LDFLAGS_RELEASE_dynamic=$(LDFLAGS_COMMON_dynamic) /implib:"$(OUTDIR)\$(RELEASE_NAME).$(STEXT)" 
   
+  RESFILE=$(RELEASE_OBJDIR_dynamic)$(PATH_SEP)stlport.res
+--- 95,101 ----
+  CXXFLAGS_RELEASE_dynamic=$(FLAGS_COMMON_dynamic) /MD $(FLAGS_NDEBUG) /Fo"$(RELEASE_OBJDIR_dynamic)\\" /Fd"$(RELEASE_OBJDIR_dynamic)\\" /YXstlport_prefix.h /Fp$(RELEASE_OBJDIR_dynamic)\stlport.pch
   
-*** misc/STLport-4.5.3/stlport/stdexcept	Tue May 22 02:50:22 2001
---- misc/build/STLport-4.5.3/stlport/stdexcept	Fri May  3 17:59:41 2002
+  LDFLAGS_RELEASE_static=$(LDFLAGS_COMMON_static)
+! LDFLAGS_RELEASE_dynamic=$(LDFLAGS_COMMON_dynamic) $(LDFLAGS_RELEASE) msvcrt.lib /implib:"$(OUTDIR)\$(RELEASE_NAME).$(STEXT)" 
+  # LDFLAGS_RELEASE_dynamic=$(LDFLAGS_COMMON_dynamic) /implib:"$(OUTDIR)\$(RELEASE_NAME).$(STEXT)" 
+  
+  RESFILE=$(RELEASE_OBJDIR_dynamic)$(PATH_SEP)stlport.res
+*** misc/STLport-4.5.3/stlport/stdexcept	Tue May 22 02:50:21 2001
+--- misc/build/STLport-4.5.3/stlport/stdexcept	Mon Jan 14 18:26:09 2002
 ***************
 *** 49,54 ****
---- 49,58 ----
+--- 49,59 ----
   
   _STLP_BEGIN_NAMESPACE
   
++ /* maybe limit to gcc 3.x */
 + #if defined( __GNUC__)
 + #define _STLP_NOTHROW_INHERENTLY throw()
 + #endif
@@ -121,70 +174,74 @@
   class _STLP_CLASS_DECLSPEC __Named_exception : public _STLP_EXCEPTION_BASE {
   public:
     __Named_exception(const string& __str) 
-*** misc/STLport-4.5.3/stlport/stl/_hash_map.h	Sun Feb  3 00:12:04 2002
---- misc/build/STLport-4.5.3/stlport/stl/_hash_map.h	Fri May  3 17:59:41 2002
+*** misc/STLport-4.5.3/stlport/config/stl_gcc.h	Thu Jun  6 22:53:04 2002
+--- misc/build/STLport-4.5.3/stlport/config/stl_gcc.h	Thu Jun  6 22:53:56 2002
 ***************
-*** 183,191 ****
+*** 226,233 ****
   
-    _Tp& operator[](const key_type& __key) {
-      iterator __it = _M_ht.find(__key);
-!     return (__it == _M_ht.end() ? 
-! 	    _M_ht._M_insert(_value_type(__key, _Tp())).second : 
-! 	    (*__it).second );
-    }
-  
-    size_type count(const key_type& __key) const { return _M_ht.count(__key); }
---- 183,192 ----
-  
-    _Tp& operator[](const key_type& __key) {
-      iterator __it = _M_ht.find(__key);
-!     if (__it == _M_ht.end()) 
-! 	    return _M_ht._M_insert(_value_type(__key, _Tp())).second;
-!     else
-! 	    return (*__it).second;
-    }
-  
-    size_type count(const key_type& __key) const { return _M_ht.count(__key); }
-*** misc/STLport-4.5.3/stlport/config/stl_gcc.h	2002-01-10 19:41:58.000000000 +0100
---- misc/build/STLport-4.5.3/stlport/config/stl_gcc.h	2002-10-07 16:49:34.000000000 +0200
-***************
-*** 248,254 ****
-  #     define _STLP_HAS_NO_EXCEPTIONS  1
-  #   endif
+  # if (__GNUC__ >= 3)
+  
+! #  define _STLP_NATIVE_INCLUDE_PATH ../g++-v3
+! #  define _STLP_NATIVE_OLD_STREAMS_INCLUDE_PATH ../g++-v3/backward
   
-! # if (__GNUC__ >= 3)
+  # elif (__GNUC_MINOR__ < 8)
   
-  #  define _STLP_NATIVE_INCLUDE_PATH ../g++-v3
-  #  define _STLP_NATIVE_OLD_STREAMS_INCLUDE_PATH ../g++-v3/backward
---- 248,278 ----
-  #     define _STLP_HAS_NO_EXCEPTIONS  1
+--- 226,233 ----
+  
+  # if (__GNUC__ >= 3)
+  
+! #  define _STLP_NATIVE_INCLUDE_PATH ../g++-v3
+! #  define _STLP_NATIVE_OLD_STREAMS_INCLUDE_PATH ../g++-v3/backward
+  
+  # elif (__GNUC_MINOR__ < 8)
+  
+***************
+*** 285,298 ****
+  # if defined(__DJGPP)
+  #   define _STLP_NATIVE_INCLUDE_PATH ../lang/cxx
+  # elif (__GNUC__ >= 3) || (__GNUC_MINOR__ >= 97)
+! #   define _STLP_NATIVE_INCLUDE_PATH ../include/g++-v3
+  # elif ((__GNUC_MINOR__ >= 95 && __GNUC_MINOR__ < 97) && !( defined (__FreeBSD__) || defined (__NetBSD__) || defined(__sgi) ) )
+! #   define _STLP_NATIVE_INCLUDE_PATH ../g++-3
+  # elif (__GNUC_MINOR__ > 8) && (__GNUC_MINOR__ < 95) && (__GNUC__ < 3) && !defined( __Lynx__ )
+  // this really sucks, as GNUpro does not really identifies itself, so we have to guess 
+  // depending on a platform
+  #   ifdef __hpux
+! #    define _STLP_NATIVE_INCLUDE_PATH ../g++-3
+  #   else
+  #    define _STLP_NATIVE_INCLUDE_PATH ../g++-2
+  #   endif
+--- 285,298 ----
+  # if defined(__DJGPP)
+  #   define _STLP_NATIVE_INCLUDE_PATH ../lang/cxx
+  # elif (__GNUC__ >= 3) || (__GNUC_MINOR__ >= 97)
+! #   define _STLP_NATIVE_INCLUDE_PATH ../include/g++-v3
+  # elif ((__GNUC_MINOR__ >= 95 && __GNUC_MINOR__ < 97) && !( defined (__FreeBSD__) || defined (__NetBSD__) || defined(__sgi) ) )
+! #   define _STLP_NATIVE_INCLUDE_PATH ../g++-v32
+  # elif (__GNUC_MINOR__ > 8) && (__GNUC_MINOR__ < 95) && (__GNUC__ < 3) && !defined( __Lynx__ )
+  // this really sucks, as GNUpro does not really identifies itself, so we have to guess 
+  // depending on a platform
+  #   ifdef __hpux
+! #    define _STLP_NATIVE_INCLUDE_PATH ../g++-v3
+  #   else
+  #    define _STLP_NATIVE_INCLUDE_PATH ../g++-2
   #   endif
+*** misc/STLport-4.5.3/src/common_rules.mak	Tue Sep 10 23:41:05 2002
+--- misc/build/STLport-4.5.3/src/common_rules.mak	Tue Sep 10 23:39:45 2002
+***************
+*** 84,90 ****
+  	$(RM) $(OUTDIR)/$(STLDEBUG_NAME).$(DYNEXT)
+  	$(RM) $(OUTDIR)/$(DEBUG_NAME).$(DYNEXT)
+  	ln -s $(RELEASE_DYNLIB) $(OUTDIR)/$(RELEASE_NAME).$(DYNEXT)
+! 	-ln -s $(DEBUG_DYNLIB) $(OUTDIR)/$(DEBUG_NAME).$(DYNEXT)
+  	ln -s $(STLDEBUG_DYNLIB) $(OUTDIR)/$(STLDEBUG_NAME).$(DYNEXT)
   
-! # define _STLP_CONCAT(X,Y) X Y
-! #  define _STLP_GCC_VERSUFFIX __GNUC__.__GNUC_MINOR__
-! # if (defined __GNUC_PATCHLEVEL__) && (__GNUC_PATCHLEVEL__ > 0)
-! #  define _STLP_GCC_VERSION \
-! 	(__GNUC__ * 10000 + __GNUC_MINOR__ * 100 + __GNUC_PATCHLEVEL__)
-! # else
-! #  define _STLP_GCC_VERSION \
-! 	(__GNUC__ * 10000 + __GNUC_MINOR__ * 100 )
-! # endif
-! 
-! # if _STLP_GCC_VERSION >= 30101
-! // gcc >= 3.1.1 - /usr/include/c++/<gcc_version>/
-! 
-! #  define _STLP_NATIVE_INCLUDE_PATH _STLP_CONCAT(../../include/c++/,_STLP_GCC_VERSUFFIX)
-! #  define _STLP_NATIVE_OLD_STREAMS_INCLUDE_PATH \
-!         _STLP_CONCAT(../../include/c++/,_STLP_CONCAT(_STLP_GCC_VERSUFFIX,/backward))
-! 
-! # elif (__GNUC__ == 3) && (__GNUC_MINOR__ == 1)
-! // gcc 3.1 - /usr/include/g++-v31/
-! 
-! #  define _STLP_NATIVE_INCLUDE_PATH ../g++-v31
-! #  define _STLP_NATIVE_OLD_STREAMS_INCLUDE_PATH ../g++-v31/backward
-! 
-! # elif (__GNUC__ == 3) && (__GNUC_MINOR__ == 0)
-! // gcc 3.0 - /usr/include/g++-v3/
+  install_unix :
+--- 84,90 ----
+  	$(RM) $(OUTDIR)/$(STLDEBUG_NAME).$(DYNEXT)
+  	$(RM) $(OUTDIR)/$(DEBUG_NAME).$(DYNEXT)
+  	ln -s $(RELEASE_DYNLIB) $(OUTDIR)/$(RELEASE_NAME).$(DYNEXT)
+! 	-ln -s $(STLDEBUG_DYNLIB) $(OUTDIR)/$(DEBUG_NAME).$(DYNEXT)
+  	ln -s $(STLDEBUG_DYNLIB) $(OUTDIR)/$(STLDEBUG_NAME).$(DYNEXT)
   
-  #  define _STLP_NATIVE_INCLUDE_PATH ../g++-v3
-  #  define _STLP_NATIVE_OLD_STREAMS_INCLUDE_PATH ../g++-v3/backward
+  install_unix :
diff -Naur OOo_1.1.3_src-orig/stlport/makefile.mk OOo_1.1.3_src/stlport/makefile.mk
--- OOo_1.1.3_src-orig/stlport/makefile.mk	2004-01-28 04:52:04.000000000 -0600
+++ OOo_1.1.3_src/stlport/makefile.mk	2004-12-15 22:27:58.015744584 -0600
@@ -72,11 +72,12 @@
 # --- Files --------------------------------------------------------
 .EXPORT : CC CXX
 .IF "$(COMID)"=="gcc3"
-	TARFILE_NAME=STLport-4.5
        .IF "$(OS)$(BUILD_OS_MAJOR)$(BUILD_OS_MINOR)"=="MACOSX103"
+               TARFILE_NAME=STLport-4.5
                PATCH_FILE_NAME=STLport-4.5-macxp-panther.patch
        .ELSE
-               PATCH_FILE_NAME=STLport-4.5.patch
+               TARFILE_NAME=STLport-4.5.3
+               PATCH_FILE_NAME=STLport-4.5.3.patch
        .ENDIF
 .ELSE			# "$(COMID)"=="gcc3"
 	.IF "$(OS)"=="MACOSX"
@@ -127,7 +128,7 @@
 		.ELIF "$(OS)"=="MACOSX"
 			BUILD_FLAGS=-f gcc-3.0-macosx.mak
 		.ELSE
-			BUILD_FLAGS=-f gcc-3.0.mak
+			BUILD_FLAGS=-f gcc.mak
 		.ENDIF
 	.ELSE # "$(COMID)"=="gcc3"
 		# MacOS X/Darwin need a special makefile
